File: Source/OldCode/Archive/2005 TVA Code Library/Version/2.0/Source/Tva.Phasors/IFrequencyValue.vb
'*******************************************************************************************************
'  IFrequencyValue.vb - Frequency value interface
'  Copyright © 2005 - TVA, all rights reserved - Gbtc
'
'  Build Environment: VB.NET, Visual Studio 2005
'  Primary Developer: James R Carroll, Operations Data Architecture [TVA]
'      Office: COO - TRNS/PWR ELEC SYS O, CHATTANOOGA, TN - MR 2W-C
'       Phone: 423/751-2827
'       Email: jrcarrol@tva.gov
'
'  Code Modification History:
'  -----------------------------------------------------------------------------------------------------
'  02/18/2005 - James R Carroll
'       Initial version of source generated
'
'*******************************************************************************************************

' This class represents the protocol independent interface of a frequency value.
Public Interface IFrequencyValue

    Inherits IChannelValue(Of IFrequencyDefinition)

    Property Frequency() As Double

    Property DfDt() As Double

    Property UnscaledFrequency() As Int16

    Property UnscaledDfDt() As Int16

End Interface

File: Source/OldCode/Archive/2007 Phasor Protocols/PhasorProtocols/IFrequencyValue.cs
//*******************************************************************************************************
//  IFrequencyValue.cs
//  Copyright © 2009 - TVA, all rights reserved - Gbtc
//
//  Build Environment: C#, Visual Studio 2008
//  Primary Developer: James R Carroll
//      Office: PSO TRAN & REL, CHATTANOOGA - MR BK-C
//       Phone: 423/751-4165
//       Email: jrcarrol@tva.gov
//
//  Code Modification History:
//  -----------------------------------------------------------------------------------------------------
//  02/18/2005 - James R Carroll
//       Generated original version of source code.
//
//*******************************************************************************************************

namespace PhasorProtocols
{
    /// <summary>
    /// Represents a protocol independent interface of a frequency value.
    /// </summary>
    public interface IFrequencyValue : IChannelValue<IFrequencyDefinition>
    {
        /// <summary>
        /// Gets or sets the floating point value that represents this <see cref="IFrequencyValue"/>.
        /// </summary>
        double Frequency { get; set; }

        /// <summary>
        /// Gets or sets the floating point value that represents the change in this <see cref="IFrequencyValue"/> over time.
        /// </summary>
        double DfDt { get; set; }

        /// <summary>
        /// Gets or sets the unscaled integer representation of this <see cref="IFrequencyValue"/>.
        /// </summary>
        int UnscaledFrequency { get; set; }

        /// <summary>
        /// Gets or sets the unscaled integer representation of the change in this <see cref="IFrequencyValue"/> over time.
        /// </summary>
        int UnscaledDfDt { get; set; }
    }
}